cmd/antrea-agent/agent.go

```diff
@@ -240,16 +240,24 @@ func run(o *Options) error {
 	egressConfig := &config.EgressConfig{
 		ExceptCIDRs: exceptCIDRs,
 	}
+	_, hostNetworkMode := config.GetHostNetworkModeFromStr(o.config.HostNetworkMode)
+	proxyAllNFTables := o.config.AntreaProxy.ProxyAll && hostNetworkMode == config.HostNetworkModeNFTables
+
+	hostNetworkAccelerationEnable := networkConfig.EnableHostNetworkAcceleration &&
+		(encapMode == config.TrafficEncapModeNoEncap || encapMode == config.TrafficEncapModeHybrid)
```

Comment:
why are we making this change? it feels like we are just adding an extra parameter to `route.NewClient`
was there an issue with the previous version?
I actually think that we could / should add the new `hostNetworkMode` to the `NetworkConfig` as well. The rationale is that this is ultimately going to be a "global" config for multiple features, and having it in the `NetworkConfig` means that we don't need to add it as a constructor parameter for multiple features (if these features already consume `NetworkConfig`).
This way we avoid growing the parameter list for `route.NewClient`